### DISCUSSION TITLE
Add a PutAndForgetAsync method to KVStore

### DISCUSSION THREAD

**Discussion by darkwatchuk** [nats-io/nats.net#603] [Ideas]
Is it worth putting one or two *Forget* methods in to the KVStore that do not care about the result coming back?  So something like this that does the publish on the underlying Connection.  This would be similar to the Redis FireAndForget?

Either that or maybe expose the underlying Connection property on the KV and JS Context objects to make it easier to extension methods?

```
   public async ValueTask PutAndForgetAsync<T>(string key, T value, INatsSerialize<T>? serializer = default, CancellationToken cancellationToken = default)
   {
       ValidateKey(key);
       await _context.Connection.PublishAsync($"$KV.{Bucket}.{key}", value, serializer: serializer, cancellationToken: cancellationToken);
   }
```

**Comment by mtmk** [comment]
that would be a design decision. the way you did it may cause the server to be overwhelmed. We probably need to think about a new bulk update for KV in general and that would be across all clients cc @ripienaar

**Reply by darkwatchuk** [reply]
Wouldn't you have the same problem with PublishConcurrentAsync which has more overhead?

**Reply by mtmk** [reply]
yes you might if you make too many concurrent requests but you can control how many since you'd also be waiting for replies. that way you can throttle the publish to server's pace.
